File: 877a.cpp
#include <bits/stdc++.h>
using namespace std;

string name[] = {"Danil", "Olya", "Slava", "Ann" ,"Nikita"};
int count;
void check(string s){
    for(auto x : name)if(s==x)count++;
}
int main()
{
    string s;
    cin>>s;
    for(int i=0; i<s.size(); i++){
        string temp = "";
        for(int j=i; j<s.size(); j++){
            temp+=s[j];
            check(temp);
        }
    }
    if(count==1) puts("YES");
    else puts("NO");
    return 0;
}

File: old_attempts/877a.py
def broken(mystr):
    
    match = ["Danil", "Olya", "Slava", "Ann", "Nikita"]
    c1 = mystr.count(match[0])
    c2 = mystr.count(match[1])
    c3 = mystr.count(match[2])
    c4 = mystr.count(match[3])
    c5 = mystr.count(match[4])
    res = [c1, c2, c3, c4, c5]
    #print(res)
    if sum(res) == 1:
        return "YES"
    else:
        return "NO"
print(broken(str(input())))